---
title: "Introduction to RICdata"
author: "Maria Dermit"
date: "`r BiocStyle::doc_date()`"
package: "`r pkg_ver('RICdata')`"
output: 
  BiocStyle::html_document:
    self_contained: yes
    toc: true
    toc_float: true
    toc_depth: 2
    code_folding: show
vignette: >
  %\VignetteIndexEntry{Introduction to RICdata}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8} 
bibliography: ["RICdata.bib"]
---

```{r required packages, echo = FALSE, warning=FALSE, results="hide"}
suppressPackageStartupMessages({
    library("BiocStyle")
    library("RICdata")
    library(QFeatures)
    library(magrittr)
    library(tidyverse)
})
```

# Introduction

`RICdata` is a data package containing the data to analyse RNA interaction 
capture from the manuscript titled: Global analysis of RNA-binding protein 
dynamics by comparative and enhanced RNA interactome capture. This paper was 
published in Nature Protocols 16,27â€“60(2021).

# Proteomics data overview

Peptide information extracted from
_PRIDE_ [PXD009789](https://www.ebi.ac.uk/pride/archive/projects/PXD009789).A 
total of 9 oligo DT capture and total cell lysate samples originated from mass 
spectrometry proteomics SILAC quantitative experiments [@Garcia-Moreno:2019].

```{r}
# Path to tabular data
WCLpeptidesfilepath<- system.file("extdata","WCL_peptides.txt", package = "RICdata")
RICpeptidesfilepath<- system.file( "extdata", "RIC_peptides.txt", package = "RICdata")

data("WCLpeptides.raw")
dim(WCLpeptides.raw)
data("RICpeptides.raw")
dim(RICpeptides.raw)
```

Indices of the columns to be used as expression values are as follow:
```{r}
j <- str_which(colnames(WCLpeptides.raw),str_c(c("Intensity.((\\D)).18_M_4",
                                             "Intensity.((\\D)).4_18_M",
                                             "Intensity.((\\D)).M_4_18"),
                                              collapse="|"))
colnames(WCLpeptides.raw)[j]

i <- str_which(colnames(RICpeptides.raw),str_c("Intensity.[H|M|L].",
                                               collapse="|"))
colnames(RICpeptides.raw)[i]
```

## Converting tabular data into QFeatures

```{r}
QWCLpeptides <- readQFeatures(WCLpeptidesfilepath, ecol = j, sep = "\t", 
                              name = "peptides", fnames = "Sequence")
QRICpeptides <- readQFeatures(RICpeptidesfilepath, ecol = i, sep = "\t", 
                              name = "peptides", fnames = "Sequence")
```
`readQFeatures` can take either file path where your` `peptides.txt` lives or
the tabular data already read.  We need to detect the position where 
__Intensity__ columns are. Note that the authors of RIC experiment gave 
different names to the oligo(dT) capture (RIC) and  WCL experiments, so we need 
to have one set of indexes for `WCL` experiment (j) and one set of indexes for
`RIC` experiment (i).

## QFeatures annotation
We can annotate with metadata our QFeatures objects. This is important as it 
defines the order and sample names of experiments. 

```{r experiment-QFeatures}
sample_names=c('hour18','hour4','mock')
QWCLpeptides$group <- paste(sample_names,rep(1:3,each=3),sep='_')
QWCLpeptides$sample <- rep(1:3, each=3)
colData(QWCLpeptides)

QRICpeptides$group <-  paste(sample_names,rep(1:3,each=3),sep='_')
QRICpeptides$sample <- rep(1:3, each=3)
colData(QRICpeptides)
```

## QFeatures filtering

We filter for contaminant proteins  and decoy database hits which are indicated
by "+" in the columns "Potential.contaminants" and "Reverse" respectively using
QFeatures-filtering functions.
```{r filtering-data}
QWCLpeptidesfiltered <- QWCLpeptides %>% 
    filterFeatures(~ Reverse == "") %>%
    filterFeatures(~ Potential.contaminant == "")

QRICpeptidesfiltered <- QRICpeptides %>% 
    filterFeatures(~ Reverse == "") %>%
    filterFeatures(~ Potential.contaminant == "")

```

## Removing unnecessary features
We can retain only rowDatanames of interest. To do this we can use the 
`QFeatures::selectRowData` function.
```{r QFeatures-selectRowData}
 
rowDataNames(QWCLpeptidesfiltered)[["peptides"]] %>% length() 
rowDataNames(QRICpeptidesfiltered)[["peptides"]] %>% length() 
 
rowvars <- c("Sequence", "Proteins", "Leading.razor.protein")
QWCLpeptidesfiltered_clean <- selectRowData(QWCLpeptidesfiltered, rowvars)
QRICpeptidesfiltered_clean <- selectRowData(QRICpeptidesfiltered, rowvars)

rowDataNames(QWCLpeptidesfiltered_clean)[["peptides"]] %>% length() 
rowDataNames(QRICpeptidesfiltered_clean)[["peptides"]] %>% length() 
```

# Annotation data overview

`RICdata` package also contains a reduced version of data contained in [ProtFeatures](https://www.huber.embl.de/users/befische/RBDmap/Rpackage/RBDmapHeLa.pdf)
[@Castello:2016]. This object is called `miniProtFeatures` and contains proteins 
sequence information. 
miniProtFeature is a list with the following objects:

- _ProtSeq_  a AAStringSet object of length 69025 with protein sequences.
- _GeneName_ a character vector with ENSEMBL gene id.
- _Symbol_ a named character vector with ENSEMBL gene symbols.

```{r}
data(miniProtFeatures)
head(miniProtFeatures$ProtSeq)
head(miniProtFeatures$GeneName)
head(miniProtFeatures$Symbol)
```

GO annotation provided in [mRNAinteractome](https://www.huber.embl.de/users/befische/mRNAinteractome/Rpackage/index.html) is included and called `ENSG2category`. 

```{r}
data(ENSG2category)
head(ENSG2category)
```

Index maps for all amino acids 4-mers to proteins is provided as `Index` object, 
and is used by the function `mapPeptides` included in RIC` package to reverse
peptides on a protein sequence database.

```{r}
data(Index)
head(Index$AAAA)
```

All these data are required to run functions in `RIC` package in order to
analyse RNA interaction capture data.


# Session Info

```{r sessioninfo}
sessionInfo()
```

