---
title: "Introduction to RICdata"
author: "Maria Dermit"
date: "`r BiocStyle::doc_date()`"
package: "`r pkg_ver('RICdata')`"
output: 
   BiocStyle::html_document
vignette: >
  %\VignetteIndexEntry{Introduction to RICdata}
  %\VignetteEngine{knitr::rmarkdown}
  %\usepackage[utf8]{inputenc}
  %\VignetteEncoding{UTF-8}
bibliography: ["RICdata.bib"]
---

```{r required packages, echo = FALSE, warning=FALSE, results="hide"}
suppressPackageStartupMessages({
    library("BiocStyle")
    library("RICdata")
    library(magrittr)
    library(tidyverse)
})
```


\pagebreak

# Introduction

`RICdata` (data from RNA interaction capture experiments) is a data package with
peptide information extracted from _PRIDE_ [PXD009789](https://www.ebi.ac.uk/pride/archive/projects/PXD009789)  for 9 oligo DT capture and total cell lysate samples originated from mass 
spectrometry proteomics SILAC quantitative experiments [@Garcia-Moreno:2019].
The package also contains a reduced version of ProtFeatures of [ProtFeatures](https://www.huber.embl.de/users/befische/RBDmap/Rpackage/RBDmapHeLa.pdf)
[@Castello:2016].
`miniProtFeatures` which contains the sequence information of the proteins. 
miniProtFeature is a list with the following objects.
- _ProtSeq_  a AAStringSet object of length 69025 with protein sequences.
- _GeneName_ a character vector with ENSEMBL gene id.
- _Symbol_ a named character vector with ENSEMBL gene symbols.
In addition the GO annotation as provided [here](https://www.huber.embl.de/users/befische/mRNAinteractome/Rpackage/index.html)
is included as `ENSG2category`. 
Index maps all amino acids 3-mers to proteins is provided as `ENSG2category`.
It is used by the function `mapPeptides` to reverse map peptides on
a protein sequence database.
All these data are required to run functions in `RIC` package.

# Session Info

```{r sessioninfo}
sessionInfo()
```

